src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -329,14 +344,48 @@ public void migrationZip(HttpServletRequest request,
 	 */
 	private String migration_call(HttpServletRequest request, HttpServletResponse response, String target, String remoteUser)
 	{
+		// we need to do series checks to make sure the migration request is valid
+		// 1. check if missing site_id or tool_id attribute
+		Map<String, String[]> parameterMap = request.getParameterMap();
+		String siteId = parameterMap.get("site_id")[0];
+		String toolId = parameterMap.get("tool_id")[0];
+		if (siteId == null || siteId.isEmpty() || toolId == null || toolId.isEmpty())
+		{
+			return  "Migration request missing required parameter: site_id, or tool_id";
```

Comment:
How will this error be surfaced to user? It looks like the error-string is ignored for zip file download and appended to a URL for a Box upload